modules/core/src/main/java/org/apache/ignite/dump/DumpReader.java

```diff
@@ -109,17 +116,26 @@ public DumpReader(DumpReaderConfiguration cfg, IgniteLogger log) {
 
                 AtomicBoolean skip = new AtomicBoolean(false);
 
-                Map<Integer, Set<Integer>> groups = cfg.skipCopies() ? new HashMap<>() : null;
+                Map<Integer, Set<Integer>> grpToParts = cfg.skipCopies() ? new HashMap<>() : null;
+
+                int partsCnt = grpToNodes.entrySet().stream().mapToInt(e ->
+                        (int)e.getValue().stream()
+                            .flatMap(node -> dump.partitions(node, e.getKey()).stream())
+                            .filter(part -> grpToParts == null || grpToParts.computeIfAbsent(e.getKey(), x -> new HashSet<>()).add(part))
+                            .count())
+                    .sum();
 
-                if (groups != null)
-                    grpToNodes.keySet().forEach(grpId -> groups.put(grpId, new HashSet<>()));
+                AtomicInteger partsProcessed = new AtomicInteger(0);
+
+                if (grpToParts != null)
+                    grpToParts.clear();
```

Comment:
Looks weird that you fill this collection in 124L, clear here, and fill again in 138L.